pkg/cmd/server/bootstrappolicy/all_test.go

```diff
@@ -0,0 +1,121 @@
+package bootstrappolicy
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const osClusterRoleAggregationPrefix = "system:openshift:"
+
+// this map must be manually kept up to date as we make changes to aggregation
+// we hard code this data with no constants because we cannot change the underlying values
+var expectedAggregationMap = map[string]sets.String{
+	"admin": sets.NewString("system:openshift:aggregate-to-admin", "system:aggregate-to-admin"),
+	"edit":  sets.NewString("system:openshift:aggregate-to-edit", "system:aggregate-to-edit"),
+	"view":  sets.NewString("system:openshift:aggregate-to-view", "system:aggregate-to-view"),
+}
+
+func TestPolicyAggregation(t *testing.T) {
```

Comment:
The test has ±100 lines, 3 nested `for` loops and many statements that do something. Unfortunately, the testing logic isn't obvious and I'm curious, could the test be improved? Even the fact that you have to put some comments is a hint that probably some functions could be extracted.